### ISSUE TITLE
Azure Monitor values 

### ISSUE THREAD

**Issue by migel0** [grafana/grafana#26828]
**What happened**:
 
After update from 6.x version to 7.x ( 7.0 , 7.1 and 7.2 tested ) some  value data from Azure Monitor Datasource are detected as  Unspecified. Specifically data from Telegraf agent.

The picture: Telegraf Agent sending metric to Azure Monitor  from VM and VMScaleset, then using Azure Monitor Datasource for  show the data.
Over Grafana prior versions, 6x this not happens.

![grafana_value2](https://user-images.githubusercontent.com/42206588/89525967-531c4400-d7e7-11ea-9165-445100955f92.png)
![grafana_value](https://user-images.githubusercontent.com/42206588/89525981-56173480-d7e7-11ea-8fd5-9536c9257b61.png)

Another inconvenient is that overwrite the value in  "yaxes" context, field context , or "overwrite" context  dont work.

- On the y axes section the values arent saved:

```json
  "yaxes": [
    {
      "label": null,
      "show": true,
      "logBase": 1,
      "min": null,
      "max": null,
      "format": "Unspecified",  // Cant save this value, dont work
      "$$hashKey": "object:829"
    },

```

-  Field context:
Overwrite the format value here dont work:
```json
    "fieldConfig": {
      "defaults": {
        "custom": {},
        "unit": "percent"
      }
```
- Overwrite section 
The instances on the scaleset are ephimeral so the matcher section work if i put  a value but dont work even with regex expression.

```json
      "overrides": [
                {
        "matcher": {
          "id": "byName",
          "options": "used" //   try regex here
        },
        "properties": [
          {
            "id": "unit",
            "value": "percent"
          }
        ]
      }]
```

**How to reproduce it (as minimally and precisely as possible)**:

-Send metrics with telegraf agent to Azure monitor
- Try  to visualize them in Grafana using Azure Monitor Datasource

**Anything else we need to know?**:

- Grafana version: 7.1
- OS Grafana is installed on: Ubuntu
- Grafana plugins: Default core instalation plugings

**Comment by kylebrandt** [comment]
If I understand correctly:

 1. I will change the datasource so it doesn't set the unit if it is "unspecified" so overrides would not be need in that case particular case.
 2. For overrides to be useful, we really need multi-field matcher (a regex would likely suffice for now): https://github.com/grafana/grafana/issues/26831

**Comment by migel0** [comment]
You have understood perfectly. thanks you.

I have just test Grafana v7.0.2 (d3c725e4ff) ( dockerhub version ) and its not affected


> If I understand correctly:
> 
> 1. I will change the datasource so it doesn't set the unit if it is "unspecified" so overrides would not be need in that case particular case.
> 2. For overrides to be useful, we really need multi-field matcher (a regex would likely suffice for now): #26831

**Comment by gfoligna** [comment]
Just FYI: I can confirm this is also happening on Cloud Enterprise v7.1.3.

Thanks @kylebrandt!

**Comment by kylebrandt** [comment]
Fixed in 7.1.5 (actually 7.1.4, but that had a bug with this so would want .5).  There are also field overrides that can match multiple columns now.
